### DISCUSSION TITLE
Compiler strict mode check for unused function arguments with `else` keyword

### DISCUSSION THREAD

**Discussion by ethanjli** [open-policy-agent/community#358] [OPA and Rego] [answered]
Hi! I'm trying out OPA v0.50.0 with strict mode enabled on the compiler, and I wanted to check if I should expect to be getting an error for an unused parameter in certain situations with functions using the `else` keyword, or if that error is unintended behavior in such situations. Specifically, here is an example of a function where I'm getting such an error in v0.50.0: https://play.openpolicyagent.org/p/OESvMjvfTg (the Rego playground is still running on v0.49.0 so it doesn't report the error). If I'm understanding the situation correctly, it appears that the error is reported because the `subject` argument isn't used in the `else = false` branch of the function definition:
```rego
is_authenticated(subject) {
	subject.authenticated
	subject.identity != ""
} else = false # opa v0.50.0 emits the error on this line; the error isn't emitted if `else = false` is removed
```

If this kind of situation is now supposed to trigger an unused variable error starting in v0.50.0, are there any workarounds in v0.50.0 which would allow me to simultaneously:
1. Continue using the compiler in strict mode (ideally still with errors about unused function params), and
2. Define functions which default to `false` rather than `undefined`

When searching about topics related to this behavior, so far I've found:
- https://github.com/open-policy-agent/opa/issues/2445

**Comment by ashutosh-narkar** [comment, marked as answer]
@ethanjli honoring a function with a default value would be a workaround. Also something like below:

```rego
is_authenticated(subject) {
        subject.authenticated
        subject.identity != ""
} else = false {
        _ := subject
}
```

But this seems like a bug. Can you please open an issue with these details?

**Reply by ethanjli** [reply]
I will file an issue (update: the issue is now opened, at https://github.com/open-policy-agent/opa/issues/5758). Thanks for your guidance on this!
